content/en/blogs/install-nfs-server-client-for-kubesphere-cluster.md

```diff
@@ -0,0 +1,347 @@
+---
+title: 'Use KubeKey to Set up a Kubernetes and KubeSphere Cluster with NFS Storage'
+keywords: Kubernetes, KubeSphere, NFS, KubeKey
+description: Set up your NFS server and use it to provide externals storage for your Kubernetes cluster.
+tag: 'KubeSphere, storage, NFS, KubeKey, installation'
+createTime: '2021-03-02'
+author: 'Sherlock'
+snapshot: 'https://ap3.qingstor.com/kubesphere-website/docs/KubeKey-lightweight-installer.png'
+---
+
+In my previous articles, I talked about how to use [KubeKey](https://github.com/kubesphere/kubekey) to set up and scale a Kubernetes cluster. As you may already know, KubeKey can do way more than that. You can also use KubeKey to install [KubeSphere](https://kubesphere.io/), a container platform running on top of Kubernetes with streamlined DevOps workflows, unified multi-cluster management and more. Besides, KubeKey is able to install cloud-native add-ons by Chart or YAML files.
+
+Among other things, storage represents one of, if not the most important element as you set up a Kubernetes cluster. Kubernetes itself supports multiple storage solutions, such as NFS-client, Ceph CSI and GlusterFS (in-tree). In this article, I am going to show you how to use KubeKey to create a Kubernetes and KubeSphere cluster with [NFS-client Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client) providing external storage.
+
+## What is NFS
+
+Network File System (NFS) provides you with remote access to files and directories on a server machine. It allows multiple clients to access shared resources over a network.
+
+You can install [NFS-client Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client) as a storage plugin for your Kubernetes cluster while you must configure an NFS server beforehand. Here is how it works: an NFS client mounts a directory on the server machine so that files residing on the NFS server are accessible to the NFS client. The directory, which you create on the server and share with your client, is also called an **exported directory**. To ensure that your client can access the directory smoothly, you must grant the access to them with several commands which I will show you later.
+
+At the same time, to make sure your client machines can use the NFS storage service, a component called `nfs-common` needs to be installed on all the client machines.
```

Comment:
on debian based linux (including ubuntu), the package is `nfs-common`, on centos based linux(including rhel), the package is `nfs-utils`. Their functionalities are similar: providing essential nfs client libraries and utilities.

Reply:
Changed